packages/web3-eth-contract/test/integration/contract_methods.test.ts

```diff
@@ -69,11 +75,32 @@ describe('contract', () => {
 			});
 
 			describe('revert handling', () => {
-				it('should returns the expected revert reason string', async () => {
-					return expect(contractDeployed.methods.reverts().call()).rejects.toThrow(
-						'REVERTED WITH REVERT',
-					);
-				});
+				itIf(getSystemTestBackend() === 'geth')(
+					'should returns the expected revert reason string',
+					async () => {
+						let error: ContractExecutionError | undefined;
+						try {
+							await contractDeployed.methods.reverts().call();
+						} catch (err: any) {
+							error = err;
+						}
+
+						// eslint-disable-next-line jest/no-standalone-expect
+						expect(error).toBeDefined();
+						// eslint-disable-next-line jest/no-standalone-expect
+						expect(error?.innerError.message).toContain('REVERTED WITH REVERT');
+					},
+				);
+
+				itIf(getSystemTestBackend() === 'ganache')(
+					'should returns the expected revert reason string',
+					async () => {
+						// eslint-disable-next-line jest/no-standalone-expect
+						return expect(contractDeployed.methods.reverts().call()).rejects.toThrow(
+							'REVERTED WITH REVERT',
+						);
+					},
+				);
 			});
```

Comment:
I think it is important to make our libarary should behave the same for all providers.